### DISCUSSION TITLE
Scrolling table shifts layout and causes horizontal scroll

### DISCUSSION THREAD

**Discussion by a47ae** [floating-ui/floating-ui#1388] [Q&A] [answered]
If one implements a scrolling table by using a wrapping div like this
```html
<div style="overflow-x: scroll">
  <table>
    ...
  </table>
</div>
```

And places a popper element in it, the body element get a horizontal overflow and scrolls. I think this is due to the popper element having a transform and transforms breaking out of the normal document flow. 

![Bildschirmfoto 2021-10-20 um 14 37 18](https://user-images.githubusercontent.com/10907455/138093965-99f051e3-8bbd-4191-83b4-9dbfb491c8ea.png)

To illustrate what I mean I created this Codesandbox: https://codesandbox.io/s/quizzical-liskov-ig6mw
Set the viewport to mobile and the whole page should have a horizontal scroll.

I tried to solve this with the `preventOverflow` modifier but sadly had no success. Does anyone know how one could solve this, or is this even considered a bug in popperjs?

**Comment by FezVrasta** [comment, marked as answer]
I don't quite understand what's the issue, may you clarify it please?

**Reply by a47ae** [reply]
Hey, thank you for the quick response and sure I'll try to be more specific.

If you take a look at the Codesandbox you see that the containing `<div>` is set to a `width` of `100vw` and the `<div>` wrapping the table has an `overflow-x: scroll` style. So if the table is larger in width than the viewport the table should scroll and no overflow should occur.

What happens if you place a popper element into a table row and the element is outside the viewport, then the whole page scrolls horizontally. Expected behaviour would be that the table scroll, but right now the table scrolls and the body is wider than `100vw`. 

I think this is due to the popper element having a `transform` property. If you have a transformed element it breaks out of the normal document flow and `overflow` gets ignored.

**Reply by FezVrasta** [reply]
Ok thanks, are you referring to this situation then?

![CleanShot 2021-10-20 at 15 08 38@2x](https://user-images.githubusercontent.com/5382443/138098970-93ab47c7-4f80-43b3-b8e1-dd0029808fc0.jpg)

This happens because Popper is set, by default, to keep button and popper elements next to each other in any case.

You can change this behavior setting `preventOverflow` `tether` option to false.

```js
createPopper(button, tooltip, {
  placement: "bottom",
  strategy: "absolute",
  modifiers: [{
    name: 'preventOverflow',
    options: {tether: false }
  }]
});
```

https://popper.js.org/docs/v2/modifiers/prevent-overflow/#tether

**Reply by a47ae** [reply]
Wow, that was quick! This was exactly what I was looking for.
Thank you very much for your answer. :)
